src/main/java/com/hedvig/claims/web/InternalController.java

```diff
@@ -453,4 +457,110 @@ public ActiveClaimsDTO getActiveClaims(@PathVariable String userId) {
     commandBus.sendAndWait(new UpdateEmployeeClaimStatusCommand(dto.getClaimId(), dto.isCoveringEmployee()));
     return ResponseEntity.accepted().build();
   }
+
+  @PostMapping("claimFiles")
+  public ResponseEntity<Void> uploadClaimsFiles(@RequestBody ClaimsFilesUploadDTO dto) {
+
+    dto.getClaimsFiles().stream().forEach(claimFile -> {
+      commandBus.sendAndWait(new UploadClaimFileCommand(
+        claimFile.getClaimFileId(),
+        claimFile.getBucket(),
+        claimFile.getKey(),
+        claimFile.getClaimId(),
+        claimFile.getContentType(),
+        claimFile.getData(),
+        claimFile.getFileName(),
+        claimFile.getImageId(),
+        claimFile.getMetaInfo(),
+        claimFile.getSize(),
+        claimFile.getUserId(),
+        claimFile.getMarkedAsDeleted(),
+        claimFile.getMarkedAsDeletedBy(),
+        claimFile.getMarkedAsDeletedAt()
+      ));
+    });
+    return ResponseEntity.noContent().build();
+  }
+
+  @GetMapping("/{claimId}/claimFiles")
+  public ResponseEntity<ClaimsFilesUploadDTO> allClaimsFiles(@PathVariable String claimId) {
+    List<UploadFile> fileUploads = fileUploadRepository.findAllByClaimsId(claimId);
+
+    List<ClaimFileDTO> claims = fileUploads.stream().map(file -> {
+      ClaimFileDTO claimFile = new ClaimFileDTO(
+        file.getId(),
+        file.getBucket(),
+        file.getKey(),
+        file.getClaimsId(),
+        file.getContentType(),
+        file.getData(),
+        file.getFileName(),
+        file.getImageId(),
+        file.getMetaInfo(),
+        file.getSize(),
+        file.getUserId(),
+        file.getMarkedAsDeleted(),
+        file.getMarkedAsDeletedBy(),
+        file.getMarkedAsDeletedAt(),
+        file.getCategory()
+      );
+        return claimFile;
+      }
+    ).collect(Collectors.toList());
+    ClaimsFilesUploadDTO claimsFiles = new ClaimsFilesUploadDTO(claims);
+    return ResponseEntity.ok(claimsFiles);
+  }
+
+  @GetMapping("/claimFile/{claimFileId}")
+  ResponseEntity<ClaimFileDTO> claimFileById(@PathVariable String claimFileId) {
+    val optionalClaimFile = fileUploadRepository.findById(claimFileId);
+
+    if(!optionalClaimFile.isPresent()) {
+      throw new RuntimeException("no claimFile found with id " + claimFileId);
+    }
+
+    val claimFile = optionalClaimFile.get();
+
+    val claimFileDto = new ClaimFileDTO(
+      claimFile.getId(),
+      claimFile.getBucket(),
+      claimFile.getKey(),
+      claimFile.getClaimsId(),
+      claimFile.getContentType(),
+      claimFile.getData(),
+      claimFile.getFileName(),
+      claimFile.getImageId(),
+      claimFile.getMetaInfo(),
+      claimFile.getSize(),
+      claimFile.getUserId(),
+      claimFile.getMarkedAsDeleted(),
+      claimFile.getMarkedAsDeletedBy(),
+      claimFile.getMarkedAsDeletedAt(),
+      claimFile.getCategory()
+    );
+
+    return ResponseEntity.ok(claimFileDto);
+  }
+
+  @PostMapping("/{claimId}/markAsDeleted/{claimFileId}")
```

Comment:
Maybe something like this?
```suggestion
@PostMapping("/{claimId}/claimFile/{claimFileId}/markAsDeleted")
```

Reply:
Honestly i think a `DELETE` method would make more sense than `POST`

Reply:
originally had it as a delete, but as I am passing the claimFileCategory as a dto, have changed to a post as cannot pass a request body to a delete i believe

Reply:
ah yes this is true. Feels weird but i guess its _ok_. This makes an excellent argue for grpc or similar, maybe its worth looking into after all :D